src/test/unit/StrategyManagerUnit.t.sol

```diff
@@ -271,34 +458,212 @@ contract StrategyManagerUnitTests is Test, Utils {
+        strategyManager.depositIntoStrategy(strategy, token, amount);
+        cheats.stopPrank();
+    }
+
+    function test_addShares_Revert_WhenSharesIsZero() external {
+        // replace dummyStrat with Reenterer contract
+        reenterer = new Reenterer();
+        dummyStrat = StrategyBase(address(reenterer));
+
+        // whitelist the strategy for deposit
+        cheats.startPrank(strategyManager.owner());
+        IStrategy[] memory _strategy = new IStrategy[](1);
+        _strategy[0] = dummyStrat;
+        cheats.expectEmit(true, true, true, true, address(strategyManager));
+        emit StrategyAddedToDepositWhitelist(dummyStrat);
+        strategyManager.addStrategiesToDepositWhitelist(_strategy);
+        cheats.stopPrank();
+
+        address staker = address(this);
+        IStrategy strategy = dummyStrat;
+        IERC20 token = dummyToken;
+        uint256 amount = 1e18;
+
+        reenterer.prepareReturnData(abi.encode(uint256(0)));
+
+        cheats.startPrank(staker);
+        cheats.expectRevert(bytes("StrategyManager._addShares: shares should not be zero!"));
+        strategyManager.depositIntoStrategy(strategy, token, amount);
+        cheats.stopPrank();
+    }
+
+    function test_addShares_Revert_WhenDepositWouldExeedMaxArrayLength() external {
+        address staker = address(this);
+        IERC20 token = dummyToken;
+        uint256 amount = 1e18;
+        IStrategy strategy = dummyStrat;
+
+        // uint256 MAX_STAKER_STRATEGY_LIST_LENGTH = strategyManager.MAX_STAKER_STRATEGY_LIST_LENGTH();
+        uint256 MAX_STAKER_STRATEGY_LIST_LENGTH = 32;
+
+        // loop that deploys a new strategy and deposits into it
+        for (uint256 i = 0; i < MAX_STAKER_STRATEGY_LIST_LENGTH; ++i) {
+            cheats.startPrank(staker);
+            strategyManager.depositIntoStrategy(strategy, token, amount);
+            cheats.stopPrank();
+
+            dummyStrat = _deployNewStrategy(dummyToken, strategyManager, pauserRegistry, dummyAdmin);
+            strategy = dummyStrat;
+
+            // whitelist the strategy for deposit
+            cheats.startPrank(strategyManager.owner());
+            IStrategy[] memory _strategy = new IStrategy[](1);
+            _strategy[0] = dummyStrat;
+            cheats.expectEmit(true, true, true, true, address(strategyManager));
+            emit StrategyAddedToDepositWhitelist(dummyStrat);
+            strategyManager.addStrategiesToDepositWhitelist(_strategy);
+            cheats.stopPrank();
+        }
+
+        require(
+            strategyManager.stakerStrategyListLength(staker) == MAX_STAKER_STRATEGY_LIST_LENGTH,
+            "strategyManager.stakerStrategyListLength(staker) != MAX_STAKER_STRATEGY_LIST_LENGTH"
+        );
+
+        cheats.startPrank(staker);
+        cheats.expectRevert(bytes("StrategyManager._addShares: deposit would exceed MAX_STAKER_STRATEGY_LIST_LENGTH"));
```

Comment:
We've done this throughout the repo, but you don't need to case the revert reason string to bytes. I'm fine either way, but just flagging

Reply:
I'm for it, removed now https://github.com/Layr-Labs/eigenlayer-contracts/pull/293/commits/45b1fcb80c90e9b979487411b3a4caac93373d50